edgelet/edgelet-docker/src/settings.rs

```diff
@@ -149,6 +149,26 @@ fn init_agent_spec(settings: &mut Settings) -> Result<(), LoadSettingsError> {
 
     agent_labels(settings)?;
 
+    // In nested scenario, Agent image can be pulled from its parent.
+    // It is possible to specify the parent address using the keyword $upstream
+    agent_image_resolve(settings)?;
+
+    Ok(())
+}
+
+fn agent_image_resolve(settings: &mut Settings) -> Result<(), LoadSettingsError> {
+    let image = settings.agent().config().image().to_string();
+
+    if let Some(parent_hostname) = settings.parent_hostname()
+    {
+        if image.starts_with("$upstream")
+        {
+            let image_nested =  format!("{}{}", parent_hostname, &image[9..]);
```

Comment:
Is there a better way to specify the size of the string "$upstream" instead of using "9"

Reply:
You're right, I used a constant.